### PR TITLE
DAOS-14101 control: Set NVMe device state zero value to UNKNOWN

### PR DESCRIPTION
**Pull request by tanabarr**
Avoid confusion when JSON command output includes fields that have not
been set server side (e.g. device state in dmg led response) by
changing SmdDevice dev_state zero value from NORMAL to UNKNOWN to
clearly indicate the state has not been set. Also return accurate
device state in dmg storage replace nvme response. 

Features: control
Required-githooks: true

### Before requesting gatekeeper:

* [x] Two review approvals and any prior change requests have been resolved.
* [x] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [x] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [x] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [x] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Review by knard-intel — APPROVED** [review]
Minor remark on error message

File: src/mgmt/srv_query.c [hunk 1]
```diff
@@ -760,11 +760,13 @@ ds_mgmt_dev_replace(uuid_t old_dev_uuid, uuid_t new_dev_uuid, Ctl__DevManageResp
 	rc = dss_ult_execute(bio_storage_dev_replace, &replace_dev_info, NULL, NULL,
 			     init_xs_type(), 0, 0);
 	if (rc != 0) {
-		D_ERROR("Unable to create a ULT\n");
+		DL_ERROR("ULT did not complete storage_dvv_replace");
```
[the remark this answers is not in the document]
Comment (tanabarr) [reply src/mgmt/srv_query.c:763]: fixed

**Review by knard-intel — DISMISSED** [review]

**Review by knard-intel — APPROVED** [review]

File: src/proto/ctl/smd.proto [hunk 2]
```diff
@@ -76,10 +76,11 @@ message BioHealthResp {
 }
 
 enum NvmeDevState {
-	NORMAL = 0;	// Device is in a normal operational state
-	NEW = 1;	// Device is new and is not yet in-use
-	EVICTED = 2;	// Device is faulty and has been evicted
-	UNPLUGGED = 3;	// Device has been physically removed
+	UNKNOWN   = 0; // Device state is unknown, zer6o value
```
Comment (kjacque) [inline src/proto/ctl/smd.proto:79]: nit - typo
```suggestion
	UNKNOWN   = 0; // Device state is unknown, zero value
```

**Review by kjacque — APPROVED** [review]
Looks like a good improvement to me.

File: src/proto/ctl/smd.proto (hunk 2 above)
Comment (tanabarr) [reply src/proto/ctl/smd.proto:79]: fix for this applied on https://github.com/daos-stack/daos/pull/12944

**Pull request merged**